cli/dataset.go

```diff
@@ -150,3 +156,145 @@ func (c *viamClient) deleteDataset(datasetID string) error {
 	printf(c.c.App.Writer, "Dataset with ID %s deleted", datasetID)
 	return nil
 }
+
+// DatasetDownloadAction is the corresponding action for 'dataset download'.
+func DatasetDownloadAction(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	if err := client.downloadDataset(c.Path(dataFlagDestination), c.String(datasetFlagDatasetID),
+		c.Bool(datasetFlagIncludeJSONLines), c.Uint(dataFlagParallelDownloads)); err != nil {
+		return err
+	}
+	return nil
+}
+
+// downloadDataset downloads a dataset with the specified ID.
+func (c *viamClient) downloadDataset(dst, datasetID string, includeJSONLines bool, parallelDownloads uint) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	var datasetFile *os.File
+	var err error
+	if includeJSONLines {
+		datasetPath := filepath.Join(dst, "dataset.jsonl")
+		if err := os.MkdirAll(filepath.Dir(datasetPath), 0o700); err != nil {
```

Comment:
Probably a dumb question but why is this necessary? Isn't this gonna be just as necessary whether we need the jsonl or not?

Reply:
This just seems to be a paradigm used everywhere (e.g. [here](https://github.com/viamrobotics/rdk/blob/3fc61c36d9564dc85b5d908c3ebfaaa526c0a27c/cli/data.go#L462)). I'm also not super sure, but I wonder if the redundancy is to just make sure that the writer does not run into an perms error.